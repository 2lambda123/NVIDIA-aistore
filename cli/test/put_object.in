echo "0123456789" > $OBJECT_1.txt
ais create bucket $BUCKET_1

ais put $OBJECT_1.txt $BUCKET_2/object_1.txt // FAIL "Bucket with name "$BUCKET_1" does not exist."
ais put $OBJECT_1.txt cloud://$BUCKET_1/object_1.txt // FAIL "Bucket with name "cloud/$BUCKET_1" does not exist."

ais put $OBJECT_1.txt ais://$BUCKET_1/object_1.txt
ais put $OBJECT_1.txt $BUCKET_1/object_2.txt
ais ls $BUCKET_1

echo "0" > $OBJECT_1.txt
ais put $OBJECT_1.txt $BUCKET_1/object_1.txt // Bump the version of 'object_1.txt' object
ais ls $BUCKET_1

# setup files structure for next commands
for d1 in {0..4}; do for d2 in {0..4}; do mkdir -p /tmp/rdir/$d1/$d2 && echo -n "0" > /tmp/rdir/$d1/$d2/test.txt; done; done

# test range absolute paths

ais create bucket $BUCKET_2
ais put "/tmp/rdir/{0..4}/{0..4}/test.txt" $BUCKET_2 -y // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_2/tmp/rdir/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

ais put "/tmp/rdir/{0..4}/{0..4}/test.txt" $BUCKET_2 -y --trim-prefix "/tmp/rdir" // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_2/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test range relative paths

ais create bucket $BUCKET_3
cd /tmp && ais put "rdir/{0..2}/{0..2}/test.txt" $BUCKET_3 -y // IGNORE
RES=0 && for d1 in {0..2}; do for d2 in {0..2}; do ais get $BUCKET_3/rdir/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
cd /tmp && ais put "./rdir/{0..2}/{0..2}/test.txt" $BUCKET_3 -y --trim-prefix rdir // IGNORE
RES=0 && for d1 in {0..2}; do for d2 in {0..2}; do ais get $BUCKET_3/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test put * absolute paths

ais create bucket $BUCKET_4
ais put "/tmp/rdir/*.txt" $BUCKET_4 -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_4/tmp/rdir/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
ais put "/tmp/rdir/*.txt" $BUCKET_4 -y -r --trim-prefix "/tmp/rdir" // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_4/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test put * relative paths

ais create bucket $BUCKET_5
cd /tmp && ais put "rdir/*.txt" $BUCKET_5 -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_5/rdir/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
cd /tmp && ais put "rdir/*.txt" $BUCKET_5 -y -r --trim-prefix "rdir" // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais get $BUCKET_5/$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# --trim-prefix
cd /tmp && ais put "rdir/*.txt" $BUCKET_5 -y -r --trim-prefix "dir" // FAIL "incorrect --trim-prefix value"
cd /tmp && ais put "rdir/*.txt" $BUCKET_5 -y -r --dry-run --trim-prefix "/tmp/rdir" >> /dev/null // skip output, but no failure
ais put "/tmp/rdir/{0..4}/{0..4}/test.txt" $BUCKET_2 -y --trim-prefix /tmp/rdir/0 // FAIL "incorrect --trim-prefix value"

rm -rf /tmp/rdir // IGNORE
