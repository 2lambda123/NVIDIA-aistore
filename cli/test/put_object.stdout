$BUCKET_1 bucket created
PUT object_1.txt into bucket ais://$BUCKET_1
PUT object_2.txt into bucket $BUCKET_1
Name			Size	Version
object_1.txt		11B	1
object_2.txt		11B	1
PUT object_1.txt into bucket $BUCKET_1
PUT object_3.txt into bucket $BUCKET_1
Name			Size	Version
object_1.txt		2B	2
object_2.txt		11B	1
object_3.txt        11B 1
$BUCKET_2 bucket created
25
25
$BUCKET_3 bucket created
9
9
$BUCKET_4 bucket created
25
25
$BUCKET_5 bucket created
25
25
